crates/shared/src/price_estimation/trade_verifier.rs

```diff
@@ -43,34 +44,45 @@ pub trait TradeVerifying: Send + Sync + 'static {
 /// and determines a price estimate based off of that simulation.
 #[derive(Clone)]
 pub struct TradeVerifier {
+    web3: Web3,
     simulator: Arc<dyn CodeSimulating>,
     code_fetcher: Arc<dyn CodeFetching>,
     block_stream: CurrentBlockStream,
-    settlement: H160,
+    settlement: GPv2Settlement,
     native_token: H160,
     quote_inaccuracy_limit: BigRational,
+    zeroex: Option<IZeroEx>,
 }
 
 impl TradeVerifier {
     const DEFAULT_GAS: u64 = 8_000_000;
     const TRADER_IMPL: H160 = addr!("0000000000000000000000000000000000010000");
 
-    pub fn new(
+    pub async fn new(
+        web3: Web3,
         simulator: Arc<dyn CodeSimulating>,
         code_fetcher: Arc<dyn CodeFetching>,
         block_stream: CurrentBlockStream,
         settlement: H160,
         native_token: H160,
         quote_inaccuracy_limit: f64,
     ) -> Self {
+        let zeroex = match IZeroEx::deployed(&web3).await {
+            Ok(instance) => Some(instance),
+            Err(DeployError::NotFound(_)) => None,
+            Err(err) => panic!("can't find deployed zeroex contract: {err:?}"),
```

Comment:
nit: "cannot find" is covered in the arm above

```suggestion
            Err(err) => panic!("Error loading deployed IZeroEx contract: {err:?}"),
```